drgn/commands/_builtin/crash/_log.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2025 Oracle and/or its affiliates
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""Implements the crash "log" command for drgn."""
+
+import argparse
+from typing import Any
+
+from drgn import Program
+from drgn import Object
+from drgn import PlatformFlags
+
+from drgn.commands import CommandArgumentError, argument, drgn_argument
+from drgn.commands.crash import CrashDrgnCodeBuilder, crash_command
+
+from drgn import Program
+from drgn.helpers.linux.printk import print_dmesg
+
+def crash_time_stamp(prog: Program) -> None:
+    print_dmesg(timestamps="human")
+
+def crash_monotonic_time_stamp(prog: Program) -> None:
+    print_dmesg()
+
+def crash_no_time_stamp(prog: Program) -> None:
+    print_dmesg(timestamps=False)
+
+@crash_command(
+    description="Dump kernel dmesg",
+    arguments=(
+        argument(
+            "-T",
+            dest="time_stamp",
+	    action="store_true",
+            help="Dump kernel dmesg in human readable time",
+        ),
+        argument(
+            "-t",
+            dest="no_time_stamp",
+            action="store_true",
+            help="Dump kernel dmesg without timestamp",
+        ),
```

Comment:
A couple of things here:

1. It doesn't make sense to specify both `-t` and `-T`, so these should be in a `mutually_exclusive_group()`.
2. You can make use of `store_const` to set the proper `timestamps` argument directly instead of needing the conditionals below. Something like:

```suggestion
        mutually_exclusive_group(
            argument(
                "-T",
                dest="timestamps",
                action="store_const",
                const="human",
                default=True,
                help="Dump kernel dmesg in human readable time",
            ),
            argument(
                "-t",
                dest="timestamps",
                action="store_false",
                default=argparse.SUPPRESS,
                help="Dump kernel dmesg without timestamp",
            ),
        ),
```

Then all you need do is call `print_dmesg(prog, timestamps=args.timestamps)`.